tests/serialization/json/JsonSerializationWriterTests.cs

```diff
@@ -323,5 +330,311 @@ public void WritesSampleObjectValueWithUntypedProperties()
+            jsonSerializationWriter.WriteByteArrayValue(null, value);
+            var contentStream = jsonSerializationWriter.GetSerializedContent();
+            using var reader = new StreamReader(contentStream, Encoding.UTF8);
+            var serializedString = reader.ReadToEnd();
+
+            // Assert
+            Assert.Equal("\"AgQG\"", serializedString);
+        }
+
+        [Fact]
+        public void WriteByteValue_IsWrittenCorrectly()
+        {
+            // Arrange
+            byte value = 5;
+
+            using var jsonSerializationWriter = new JsonSerializationWriter();
+
+            // Act
+            jsonSerializationWriter.WriteByteValue(null, value);
+            var contentStream = jsonSerializationWriter.GetSerializedContent();
+            using var reader = new StreamReader(contentStream, Encoding.UTF8);
+            var serializedString = reader.ReadToEnd();
+
+            // Assert
+            Assert.Equal("5", serializedString);
+        }
+
+        [Fact]
+        public void WriteDateTimeOffsetValue_IsWrittenCorrectly()
+        {
+            // Arrange
+            var value = new DateTimeOffset(2024, 11, 30, 15, 35, 45, 987, TimeSpan.FromHours(3));
+
+            using var jsonSerializationWriter = new JsonSerializationWriter();
+
+            // Act
+            jsonSerializationWriter.WriteDateTimeOffsetValue(null, value);
+            var contentStream = jsonSerializationWriter.GetSerializedContent();
+            using var reader = new StreamReader(contentStream, Encoding.UTF8);
+            var serializedString = reader.ReadToEnd();
+
+            // Assert
+            Assert.Equal("\"2024-11-30T15:35:45.987+03:00\"", serializedString);
+        }
+
+        [Fact]
+        public void WriteDateValue_IsWrittenCorrectly()
+        {
+            // Arrange
+            var value = new Date(2024, 11, 30);
+
+            using var jsonSerializationWriter = new JsonSerializationWriter();
+
+            // Act
+            jsonSerializationWriter.WriteDateValue(null, value);
+            var contentStream = jsonSerializationWriter.GetSerializedContent();
+            using var reader = new StreamReader(contentStream, Encoding.UTF8);
+            var serializedString = reader.ReadToEnd();
+
+            // Assert
+            Assert.Equal("\"2024-11-30\"", serializedString);
+        }
+
+        [Fact]
+        public void WriteDecimalValue_IsWrittenCorrectly()
+        {
+            // Arrange
+            var value = 36.8m;
+
+            using var jsonSerializationWriter = new JsonSerializationWriter();
+
+            // Act
+            jsonSerializationWriter.WriteDecimalValue(null, value);
+            var contentStream = jsonSerializationWriter.GetSerializedContent();
+            using var reader = new StreamReader(contentStream, Encoding.UTF8);
+            var serializedString = reader.ReadToEnd();
+
+            // Assert
+            Assert.Equal("36.8", serializedString);
```

Comment:
Looks like runtimes before net3.0 didn't have assurances that double/float did not guarantee returning round trippable strings when formatted. https://devblogs.microsoft.com/dotnet/floating-point-parsing-and-formatting-improvements-in-net-core-3-0/

Since this is a platform specific issue that was fixed in later runtimes, I would suggest using a `#ifdef` here to have an assertion for netfx to keep the validation here.